internal/edgeapi/backend/api.go

```diff
@@ -0,0 +1,65 @@
+package backend
+
+import (
+	"context"
+
+	"github.com/project-flotta/flotta-operator/models"
+)
+
+type Notification struct {
+	DeviceID  string
+	Namespace string
+	Heartbeat *models.Heartbeat
+	Retry     int32
+}
+
+//go:generate mockgen -package=backend -destination=mock_heartbeat-handler.go . HeartbeatHandler
+type HeartbeatHandler interface {
+	Process(ctx context.Context, notification Notification) (bool, error)
+}
+
+// Backend represents API provided by data storage service to support edge device lifecycle.
+type Backend interface {
+	// ShouldEdgeDeviceBeUnregistered responds with true, when the device identified with name and namespace should be
+	// instructed to execute de-registration procedure
+	ShouldEdgeDeviceBeUnregistered(ctx context.Context, name, namespace string) (bool, error)
```

Comment:
Semantic issue, but I think it would be more correctly to use the verb `Deregister` than `Unregister`:
https://grammarhow.com/unregister-vs-deregister/
Personally I would avoid `Should` in the interface, perhaps `CanEdgeDeviceBeDeregistered` or `IsDeviceDeregistrable`... what do you think?

Reply:
I kinda agree - `should` smells like a test method. As for other names - see https://github.com/project-flotta/flotta-operator/pull/236#discussion_r884143325